front/src/TweetDetail.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { TweetType } from './utils/types';
+import useCallApi from './utils/api';
+import TweetTree from './TweetTree';
+import SingleTweet from './SingleTweet';
+
+import List from '@mui/material/List';
+import ListItem from '@mui/material/ListItem';
+
+const TweetDetail: React.FC = () => {
+    const callApi = useCallApi();
+    const params = useParams();
+    const [tweet, setTweet] = useState<TweetType>();
+    const [replyTweetsList, setReplyTweetsList] = useState<Array<TweetType>>([]);
+
+    useEffect(() => {
+        getTweet()?.then(setTweet);
+        getReplys()?.then(setReplyTweetsList);
+    }, [params.tweetId]);
+
+    function getTweet(): Promise<TweetType> | undefined {
+        return callApi('http://localhost:5000/api/zetter/tweet/' + params.tweetId);
+    }
+
+    function getReplys(): Promise<Array<TweetType>> | undefined {
+        return callApi('http://localhost:5000/api/zetter/replys/' + params.tweetId);
+    }
+
+    function getAndSetTweets(): void {
+        getTweet()?.then(setTweet);
+        getReplys()?.then(setReplyTweetsList);
+    }
+
+    function updateFavoriteState(tweet: TweetType): void {
+        if (tweet.isFavorite) {
+            // favから削除する
+            callApi('http://localhost:5000/api/zetter', {
+                method: 'PATCH',
+                body: JSON.stringify({ tweet: tweet, order: 'delete' }),
+            })?.then((data) => {
+                setTweet(data);
+            });
+        } else {
+            // favに追加する
+            callApi('http://localhost:5000/api/zetter', {
+                method: 'PATCH',
+                body: JSON.stringify({ tweet: tweet, order: 'add' }),
+            })?.then((data) => {
+                setTweet(data);
+            });
+        }
+    }
+
+    function updateReplyFavoriteState(reply: TweetType): void {
+        if (reply.isFavorite) {
+            callApi('http://localhost:5000/api/zetter', {
+                method: 'PATCH',
+                body: JSON.stringify({ tweet: reply, order: 'delete' }),
+            })?.then((data) => {
+                let result = replyTweetsList.map(function (value: TweetType): TweetType {
+                    if (reply.id === value.id) {
+                        return data;
+                    } else {
+                        return value;
+                    }
+                });
+                setReplyTweetsList(result);
+            });
```

Comment:
```suggestion
            })?.then((newReply) => {
                let newReplies = replyTweetsList.map(reply =>  {
                    if (newReply.id === reply.id) {
                        return newReply;
                    } else {
                        return reply;
                    }
                });
                setReplyTweetsList(result);
            });
```
1. 名前がdataとかresultだと何を指してるかわかりにくくて読みにくかった　具体的な名前のほうが良いと思う
2. 他のところではアロー関数使ってるので、ここも合わせていこう